### ISSUE TITLE
Error in typing and text display when typing in right-to-left languages such as Persian (Farsi), Arabic, etc.

### ISSUE THREAD

**Issue by aaa1386** [freeplane/freeplane#1415]
Some users use Farsi or Arabic or other languages that operate on the basis of right to left in the mind map.
  Some errors occur in different areas of Freeplane, such as note panel or node editing.
For example, the home and end keys do not work well when editing. Or numerical list or bullet list option
In the video below, the bugs are shown
In the pictures, the correct mode is shown in right-to-left languages in Microsoft Word software.

⬇️⬇️⬇️A mind map with sample characters of a language that is based on right to left

[example freeplane file.zip](https://github.com/freeplane/freeplane/files/12658068/example.freeplane.file.zip)

❗️❗️❗️❗️❗️To better understand the errors, I recommend installing the Arabic or Farsi (Persian) language on Windows and testing.

In this video, when I am typing, I only put a space after each word, See irregular movements of words⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️

https://github.com/freeplane/freeplane/assets/38586259/63ec18c4-0624-4777-81fc-9a0fb97e10bd

Other errors in typing and display⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️⬇️

https://github.com/freeplane/freeplane/assets/38586259/8be10e0c-1db8-4739-b117-a8d38dc3b3dc

The correct mode is shown in right-to-left languages in Microsoft Word software.☑️☑️☑️☑️☑️☑️☑️☑️☑️  ⬇️⬇️⬇️⬇️⬇️

![image](https://github.com/freeplane/freeplane/assets/38586259/9900f315-ead4-40aa-826a-a09d78ad3650)
![image](https://github.com/freeplane/freeplane/assets/38586259/3efc3dcf-b55f-4f93-967c-25f505126216)

**Comment by theelectricsoul** [comment]
@ajaberi1360 
I tested it. The cursor position is jumping in freeplane but also in word.

Have you tried changing the language in Freeplane?

**Comment by aaa1386** [comment]
Yes, I changed the language of the program from the settings section to Arabic, which is based on right to left, but the errors are still there.
In fact, these errors are related to the difference between right-to-left or left-to-right languages. The Freeplane software focuses mainly on left-to-right languages such as English. I hope the developers of this software will correct these errors

**Comment by dpolivaev** [comment]
If Freeplane does not fully support right-to-left languages, it is neither a bug nor an error but a not implemented feature.

**Comment by aaa1386** [comment]
Thank you .
I created this thread to help make the good Freeplane software more complete.
If it is not possible to fix the reported defects, it is not a problem, I use the software in the same situation.
But if you can correct the mentioned items, the software will be more complete and users are more comfortable to use
Thank you again for your efforts

**Comment by theelectricsoul** [comment]
@ajaberi1360 if dpolivaev personally adds a "make sense" tag to a request, you are good :) And he did :)   From my experience we just have to wait. 

Making it work for arabic languages seems like a very sensible goal to reach, so ... I am confident it will be implemented. Depends on the effort.

**Comment by aaa1386** [comment]
@theelectricsoul 
I checked that 25 countries in the world speak Arabic. The statistics are significant.
  There are other languages such as Farsi that are based on right to left.
Maybe there are other languages, I didn't check more than that.
  
I don't know whether Freeplane developers pay attention to such languages or not. Whatever they decide, I respect their decision.

**Comment by euu2021** [comment]
In sourceforge, it's possible to see statistcs and maps by number of downloads: 
https://sourceforge.net/projects/freeplane/files/stats/map

Saudi Arabia is in the all time Top10:
![image](https://github.com/freeplane/freeplane/assets/77707706/6fcb2fe7-f083-43ba-9d59-1c057650de52)

**Comment by dpolivaev** [comment]
I care about everyone, so I checked what can be done for better support of right-to-left languages.
I found that this option can be configured for each node, detail, or note, but can not be changed inside one node, detail, or note.

Therefore I am going to add a node format checkbox "right-to-left", with a default value coming from the language set as system locale and also configurable as a Freeplane user setting.

At the moment I convert this issue to a discussion to get wider feedback from the community about this approach.
